src/codeu/chat/server/Model.java

```diff
@@ -67,6 +69,27 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final Store<User, HashSet<Uuid>> userInterestByUser = new Store<>(null);
+  private final Store<User, HashSet<Uuid>> convoInterestByUser = new Store<>(null);
+
+  public void addUserInterest(User user, User interested, UserInterests userInterests) {
+    userInterests.interestedInUsers.add(interested.id);
+    userInterestByUser.insert(user, userInterests.interestedInUsers);
```

Comment:
The implementation should be something like this:

```
userInterest = userInterestByUser.first(user.id);
if(userInterest == null) {
  userInterest = new UserInterests();
  userInterestByUser.insert(user.id, userInterest);
}
userInterest.interestedInUsers.add(interested.id);
```

The `first` function on Store is here:
https://github.com/lianasyrkett/CodeU-Summer-2017/blob/ff639447ffe6d2e2f46e8781a0ba7be10fbc1c18/src/codeu/chat/util/store/Store.java#L71